### PR TITLE
[LV][AArch64] Don't query registers for illegal scalable vector elts

### PR DESCRIPTION
**Pull request by huntergr-arm**
When trying to maximize vector bandwidth we ask TTI for the number of
registers required for a given operation. If the type of that operation
happens to be something illegal for scalable vectors (e.g.
<vscale x 4 x fp128>) then we would see a crash.

Instead, just return a default value and let the cost model reject the
invalid operation later.

### CODE REVIEW

**Review by paulwalker-arm — APPROVED** [review]

File: llvm/test/Transforms/LoopVectorize/AArch64/scalable-fp-ext-trunc-illegal-type.ll [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  br i1 %exitcond, label %for.body, label %for.exit, !llvm.loop !0
+
+for.exit:
+  ret void
+}
+
+attributes #0 = { "target-features"="+sve" vscale_range(1,16) }
+
+!0 = distinct !{!0, !1}
+!1 = !{!"llvm.loop.interleave.count", i32 1}
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/AArch64/scalable-fp-ext-trunc-illegal-type.ll:81]: is this needed?

File: llvm/test/Transforms/LoopVectorize/AArch64/scalable-fp-ext-trunc-illegal-type.ll [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  store float %trunc.out, ptr %gep.out, align 4
+  %iv.next = add nuw nsw i64 %iv, 1
+  %exitcond = icmp ult i64 %iv.next, %N
+  br i1 %exitcond, label %for.body, label %for.exit, !llvm.loop !0
+
+for.exit:
+  ret void
+}
+
+attributes #0 = { "target-features"="+sve" vscale_range(1,16) }
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/AArch64/scalable-fp-ext-trunc-illegal-type.ll:78]: Might be simpler/easier to see if using `-mattr=+sve` in the run line?

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

File: llvm/test/Transforms/LoopVectorize/AArch64/scalable-fp-ext-trunc-illegal-type.ll (hunk 1 above)
Comment (huntergr-arm) [reply llvm/test/Transforms/LoopVectorize/AArch64/scalable-fp-ext-trunc-illegal-type.ll]: I wanted to reduce the number of check lines to review; I've switched to using a flag instead to match the other change.

File: llvm/test/Transforms/LoopVectorize/AArch64/scalable-fp-ext-trunc-illegal-type.ll (hunk 2 above)
Comment (huntergr-arm) [reply llvm/test/Transforms/LoopVectorize/AArch64/scalable-fp-ext-trunc-illegal-type.ll]: done

**Pull request merged**